### ISSUE TITLE
Prompt users to delete environment when disconnecting

### ISSUE THREAD

**Issue by lostintangent** [MicrosoftDocs/vsonline#139]
While minimal, there is a cost associated with environments, even when they’re in a suspended state. As a result, we should explore whether it makes sense to prompt users to delete their environments when they disconnect from them. This would provide folks with a helpful reminder, and could be very beneficial for environments that were meant for single-use (e.g. PR reviews, prototyping, learning). 

To ensure that this behavior isn’t annoying for users that have “long running” environments, the prompt could simply have a “Don’t ask me again” button, in order to suppress it in the future. This way, new users would be reminded to clean up unneeded environments, while experienced users could disable the reminders and manually manage their environment’s lifetime.

**Comment by cmbrose** [comment]
We could also have an option in the prompt to just suspend the environment immediately rather than fully delete it.

**Comment by lostintangent** [comment]
@cmbrose Good point! Though at that point, we’d likely want to enable a suspend timeout of “Immediate” (or whatever). If you wanted to immediately suspend your environment, without deleting it, it would kind of defeat the purpose of auto-managed lifetimes, if you had to manually suspend every time you disconnected. Maybe I’ll file a separate enhancement issue to track that behavior, since I could see it being potentially useful.

**Comment by lostintangent** [comment]
@cmbrose Filed #141 as an extension of this discussion. I agree that adding a “Suspend” button to the disconnection prompt could be valuable. However, it also might be worth getting feedback on a new timeout option as well.

**Comment by fenixsolorzano** [comment]
I would love this feature, the current state of vs online doesn't support file sync between the remote environment and the local machine, I've found myself having to the following:

1. `git commit > pull > push` from the dev env
2. Close connection
3. Delete current env
4. Pull code from git
5. Possibly doing some changes before creating the dev env again
6. Create dev env again (three steps in this one)

It would be nice to have an option to shorten step 2 and 3 -> **Close and delete env**
